medusa/providers/torrent/torrent_provider.py

```diff
@@ -137,6 +137,43 @@ def _verify_download(self, file_name=None):
 
         return False
 
+    def _verify_magnet(self, file_name=None):
+        """
+        Validate Magnet file.
+
+        Check if the Magnet file exists and has a valid info_hash.
+        :param file_name: Absolute path to the Magnet file.
+        :returns: True or False
+        """
+        if not file_name or not os.path.isfile(file_name):
+            return False
+
+        magnet_uri = None
+        with open(file_name, 'r', encoding='utf-8') as fp:
+            magnet_uri = fp.read()
+
+        if self._get_info_from_magnet(magnet_uri):
+            return True
+        return False
+
+    @staticmethod
+    def _get_torrent_name_from_magnet(magnet_uri):
+        """Try to extract a torrent name from a Magnet URI."""
+        torrent_name = ''
+        try:
+            torrent_name = re.findall('dn=([^&]+)', magnet_uri)[0]
+        except Exception:
+            torrent_name = 'NO_DOWNLOAD_NAME'
+        return torrent_name
+
+    @staticmethod
+    def _get_info_from_magnet(magnet_uri):
+        """Try to extract an info_hash from a Magnet URI."""
+        info_hash = re.findall(r'urn:btih:([\w]{32,40})', magnet_uri)[0].upper()
+        if len(info_hash) == 32:
```

Comment:
What happens if `info_hash` is None or not 32 long? Maybe we should handle this.

Reply:
It extracts a hash of 32 or 40 chars long. If 32, it needs to do another translation.